classes/task/merge_user_accounts.php

```diff
@@ -0,0 +1,184 @@
+            (object)[
+                'id' => $idrecord,
+                'status' => $status,
+                'timecompleted' >= time(),
+                'log' => json_encode($log),
+            ]
+        );
+    }
+    /**
+     * Function for verifying the users to keep and remove.
+     */
+    private function verify_users_to_keep_and_remove(object $record) {
+        global $DB;
+        $returnvalue = 0;
+        $keepuserfield = $record->keepuserfield;
+        $keepuservalue = $record->keepuservalue;
+        $removeuserfield = $record->removeuserfield;
+        $removeuservalue = $record->removeuservalue;
+        // Verify user to remove.
+        $usertoremove = $DB->get_records(merge_request::TABLE_USERS, [$removeuserfield => $removeuservalue]);
+        if (is_null($record->removeuserid)) {
+            $this->removeuserid = $this->find_user_id_or_fail($record->removeuserfield, $record->removeuservalue);
+        }
+        // Verify user to keep.
+        $usertokeep = $DB->get_records(merge_request::TABLE_USERS, [$keepuserfield => $keepuservalue]);
+        if (is_null($record->keepuserid)) {
+            $this->keepuserid = $this->find_user_id_or_fail($record->keepuserfield, $record->keepuservalue);
+        }
+        return $returnvalue = 1;
+    }
+    /**
+     * Function for updating number of retries.
+     */
+    private function update_retries_in_table(int $idrecord,
+                                             int $retries) {
+        global $DB;
+        $DB->update_record(
+            merge_request::TABLE_MERGE_REQUEST,
+            (object)[
+                'id' => $idrecord,
+                'retries' => $retries,
+                'timecompleted' >= time(),
+            ]
+        );
+    }
+    /**
+     * Function for updating removeuserid.
+     */
+    private function update_removeuserid_in_table(int $idrecord,
+                                                        int $removeuserid) {
+        global $DB;
+        $DB->update_record(
+            merge_request::TABLE_MERGE_REQUEST,
+            (object)[
+                'id' => $idrecord,
+                'removeuserid' => $removeuserid
+            ]
+        );
+    }
+    /**
+     * Function for updating keepuserid.
+     */
+    private function update_keepuserid_in_table(int $idrecord,
+                                                int $keepuserid) {
+        global $DB;
+        $DB->update_record(
+            merge_request::TABLE_MERGE_REQUEST,
+            (object)[
+                'id' => $idrecord,
+                'keepuserid' => $keepuserid
+            ]
+        );
+    }
+    /**
+     * Function to find user id or fail.
+     */
+    protected function find_user_id_or_fail(int $mergerequestid, string $userfield, string $uservalue): int {
+        global $DB;
+        $user = $DB->get_records(merge_request::TABLE_USERS, [$userfield => $uservalue]);
+        if (count($usertoremove) == 0) {
```

Comment:
I made an error while providing the example code: there is no `$usertoremove` variable in this method; `$user` should be used instead. Or a better name, like `$users`, meaning that several potential users can be obtained.